tests/stub/datatypes/test_unknown_type.py

```
@@ -0,0 +1,172 @@
+import json
+from contextlib import contextmanager
+
+from nutkit import protocol as types
+from nutkit.frontend import Driver
+from tests.shared import TestkitTestCase
+from tests.stub.shared import StubServer
+
+
+class TestUnknownTypes(TestkitTestCase):
+
+    required_features = (
+        types.Feature.API_TYPE_UNKNOWN_TYPE,
+        types.Feature.BOLT_6_0,
+    )
+    bolt_version = "6.0"
+
+    def setUp(self):
+        super().setUp()
+        self._server = StubServer(9010)
+
+    def tearDown(self):
+        self._server.reset()
+        super().tearDown()
+
+    @contextmanager
+    def _started_server(self, server, script, vars_=None):
+        version_folder = "v{}".format(self.bolt_version.replace(".", "x"))
+        server.start(
+            path=self.script_path(version_folder, script),
+            vars_=vars_,
+        )
+        try:
+            yield
+        finally:
+            server.reset()
+
+    @contextmanager
+    def _driver(self, server):
+        uri = "bolt://%s" % server.address
+        auth = types.AuthorizationToken("basic", principal="", credentials="")
+        driver = Driver(self._backend, uri, auth)
+        try:
+            yield driver
+        finally:
+            driver.close()
+
+    @contextmanager
+    def _session(self, driver):
+        session = driver.session("r")
+        try:
+            yield session
+        finally:
+            session.close()
+
+    def test_unknown_type_subtests(self):
+
+        script = "echo_unknown.script"
+        for (name, minimum_protocol_major, minimum_protocol_minor, extra) in (
+            ("encrypted_value", 6, 10, None),
+            ("encrypted_value", 6, 10, {"message": "test message"}),
+            (
+                "encrypted_value", 6, 10,
+                {"message": "test message", "junk data": "junk"}
```

The way the stub server changes are, rn, this test is useless as the sever will silently swallow the extra key (see comment above).